test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,328 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/search/algorithm/all.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+struct options
+{
+    size_t const sequence_length;
+    bool const has_repeats;
+    size_t const number_of_reads;
+    size_t const read_length;
+    double const prob_insertion;
+    double const prob_deletion;
+    uint8_t const simulated_errors;
+    uint8_t const searched_errors;
+    uint8_t const strata;
+    double const stddev{0};
+    uint32_t repeats{20};
+};
+
+template <Alphabet alphabet_t>
+void mutate_insertion(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937_64 gen{seed};
+    std::uniform_int_distribution<uint8_t> dis_alpha{0, alphabet_size<alphabet_t> - 1};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    seq.insert(std::ranges::begin(seq) + random_pos(gen), alphabet_t{}.assign_rank(dis_alpha(gen)));
+}
+
+template <Alphabet alphabet_t>
+void mutate_deletion(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937_64 gen{seed};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    seq.erase(std::ranges::begin(seq) + random_pos(gen));
+}
+
+template <Alphabet alphabet_t>
+void mutate_substitution(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937_64 gen{seed};
+    std::uniform_int_distribution<uint8_t> dis_alpha_short{0, alphabet_size<alphabet_t> - 2};
+    std::uniform_int_distribution<size_t> random_pos{0, std::ranges::size(seq) - overlap};
+    alphabet_t & cbase = seq[random_pos(gen)];
+    uint8_t crank = to_rank(cbase);
+    uint8_t rrank = dis_alpha_short(gen);
+    if (rrank >= crank)
+        ++rrank;
+    cbase.assign_rank(rrank);
+}
+
```

Comment:
If I understand this correctly, you are creating a new distribution for every single mutation? If you already have an RNG in the calling scope and already pass in the seed, why not directly pass in the position that you want to change? Would reduce the lines of code and likely be a lot faster!

Reply:
So I reworked these functions, because I finally found the way to do exact number of mutation with no bias in regards to their position. (eg. Now Insertion can no longer push previous mutation outside the read (fixed read length)).